core/drivers/crypto/versal/rsa.c

```diff
@@ -0,0 +1,364 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) Foundries Ltd. 2022.
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ */
+
+#include <assert.h>
+#include <crypto/crypto_impl.h>
+#include <drvcrypt.h>
+#include <drvcrypt_acipher.h>
+#include <initcall.h>
+#include <ipi.h>
+#include <mm/core_memprot.h>
+#include <string.h>
+#include <tee/cache.h>
+
+#define RSA_MAX_PRIV_EXP_LEN	512
+#define RSA_MAX_PUB_EXP_LEN	4
+#define RSA_MAX_MOD_LEN		512
+
+static void crypto_bignum_bn2bin_pad(size_t size,
+				     struct bignum *from, uint8_t *to)
+{
+	size_t len = crypto_bignum_num_bytes(from);
+
+	crypto_bignum_bn2bin(from, to + size - len);
+}
+
+static TEE_Result do_encrypt(struct drvcrypt_rsa_ed *rsa_data)
+{
+	struct rsa_public_key *p = rsa_data->key.key;
+	struct versal_rsa_input_param *cmd = NULL;
+	struct versal_mbox_mem cmd_buf = { };
+	struct versal_mbox_mem cipher = { };
+	struct versal_mbox_mem key = { };
+	struct versal_mbox_mem msg = { };
+	struct versal_cmd_args arg = { };
+	TEE_Result ret = TEE_SUCCESS;
+	uint32_t err = 0;
+
+	switch (rsa_data->rsa_id) {
+	case DRVCRYPT_RSA_PKCS_V1_5:
+		if (rsa_data->key.n_size != 128)
+			break;
+
+		return sw_crypto_acipher_rsaes_encrypt(rsa_data->algo,
+						rsa_data->key.key,
+						rsa_data->label.data,
+						rsa_data->label.length,
+						rsa_data->message.data,
+						rsa_data->message.length,
+						rsa_data->cipher.data,
+						&rsa_data->cipher.length);
+		break;
```

Comment:
can remove he `break` instructions, here and in `do_decrypt()`, `do_rsa_sign()` and `do_rsa_verify()`.

Reply:
a lot of warnings in checkpatch which is what I was trying to avoid but I am sure you are aware of those.
added your acked-by after addressing the comments.

Reply:
checkpatch looks happy

Reply:
um weird, this is what I see on my box

```
Checking commit(s):
af23fb20f crypto: versal: RSA driver
WARNING: Possible switch case/default not preceded by break or fallthrough comment
#79: FILE: core/drivers/crypto/versal/rsa.c:54:
+       case DRVCRYPT_RSA_OAEP:

WARNING: Possible switch case/default not preceded by break or fallthrough comment
#88: FILE: core/drivers/crypto/versal/rsa.c:63:
+       case DRVCRYPT_RSA_NOPAD:

WARNING: Possible switch case/default not preceded by break or fallthrough comment
#94: FILE: core/drivers/crypto/versal/rsa.c:69:
+       case DRVCRYPT_RSASSA_PKCS_V1_5:

WARNING: Possible switch case/default not preceded by break or fallthrough comment
#171: FILE: core/drivers/crypto/versal/rsa.c:146:
+       case DRVCRYPT_RSA_OAEP:

WARNING: Possible switch case/default not preceded by break or fallthrough comment
#180: FILE: core/drivers/crypto/versal/rsa.c:155:
+       case DRVCRYPT_RSA_NOPAD:

WARNING: Possible switch case/default not preceded by break or fallthrough comment
#186: FILE: core/drivers/crypto/versal/rsa.c:161:
+       case DRVCRYPT_RSASSA_PKCS_V1_5:

WARNING: Possible switch case/default not preceded by break or fallthrough comment
#253: FILE: core/drivers/crypto/versal/rsa.c:228:
+       case TEE_ALG_RSASSA_PKCS1_V1_5_MD5:

WARNING: Possible switch case/default not preceded by break or fallthrough comment
#254: FILE: core/drivers/crypto/versal/rsa.c:229:
...
```